### ISSUE TITLE
Question about cosine similarity 

### ISSUE THREAD

**Issue by hoyden** [google-research/simclr#38]
Why did you only do dot product when calculating cosine similarity, but not divided by the magnitude of the vector？As follows:
`logits_aa = tf.matmul(hidden1, hidden1_large, transpose_b=True) / temperature`

**Comment by chentingpc** [comment]
the hiddens are l2-normalized before dot product (when `hidden_norm` is set to True), which makes it cosine similarity.

**Comment by hoyden** [comment]
@chentingpc So did you calculate the cosine similarity exactly according to the formula when you test the importance of the l2 normalization in table 5?

**Comment by Some-random** [comment]
@chentingpc I'm having a similar question. Is this **l2-normlization on hidden states** the improvements the paper talked about in Section 5.1? It looks like the l2-norm is just a substitute for dividing magnitude of vector when calculating similarity...

**Comment by chentingpc** [comment]
the l2-normalization of hidden states is equivalent to the use cosine similarity instead of simply dot product. I will revise the paper to make it more clear. feel free to let me know if it is still not clear.
